From: Ben Hutchings <ben@decadent.org.uk>
Date: Fri, 11 Jul 2014 17:41:42 +0100
Subject: blkcg: Avoid ABI change in 3.14.12

The type of blkcg_gq::refcnt was changed from int to atomic_t,
and reference counting now requires atomic operations.  None
of the in-tree users are modular and this structure isn't exposed
OOT, so we can hide this change.

--- a/block/blk-cgroup.h
+++ b/block/blk-cgroup.h
@@ -18,7 +18,9 @@
 #include <linux/seq_file.h>
 #include <linux/radix-tree.h>
 #include <linux/blkdev.h>
+#ifndef __GENKSYMS__
 #include <linux/atomic.h>
+#endif
 
 /* Max limits for throttle policy */
 #define THROTL_IOPS_MAX		UINT_MAX
@@ -105,7 +107,11 @@ struct blkcg_gq {
 	struct request_list		rl;
 
 	/* reference count */
+#ifndef __GENKSYMS__
 	atomic_t			refcnt;
+#else
+	int				refcnt;
+#endif
 
 	/* is this blkg online? protected by both blkcg and q locks */
 	bool				online;
